models/record.go

```diff
@@ -587,19 +613,20 @@ func Downcase(recs []*RecordConfig) {
 
 // CanonicalizeTargets turns Targets into FQDNs
 func CanonicalizeTargets(recs []*RecordConfig, origin string) {
+	originFQDN := dns.Fqdn(origin)
```

Comment:
```suggestion
	originFQDN := origin+"."
```

There's never a situation where the "." already exists.  If there is, that would be an API change and we'd actually want this to fail.

Reply:
Ok I will pick this up when I implement the rest of your suggestions.